### DISCUSSION TITLE
How to fill these areas

### DISCUSSION THREAD

**Discussion by Zhanlf0226** [mtex-toolbox/mtex#1130] [Ask Anything] [answered]
Dear colleagues

The grains have some white areas in my data.  I refer to the content in the " filling missing data", but it  had no effect. I think my code is error, and the white area maybe not missing data. 
![image](https://user-images.githubusercontent.com/87469624/125756457-5a101084-64b4-4b4c-ae23-fceb6ae8869e.png)
![image](https://user-images.githubusercontent.com/87469624/125756612-1f6e9b92-f5b8-4dff-9cf3-eca3b1e71c53.png)
![image](https://user-images.githubusercontent.com/87469624/125756683-8cc631e7-55c6-4bfa-a0d6-455c7bea2d11.png)

Now I want to know how to fill these white areas. 

Thanks in advance.

Zhanlf

**Comment by nyyssont** [comment]
Doing:

```
plot(ebsd)
```

should reveal if the indicated data is indexed or not, as well as which phase it has been indexed as.

**Reply by Zhanlf0226** [reply]
i can't understand. i want to fill these white areas. 
![image](https://user-images.githubusercontent.com/87469624/125759062-9f3a3cdf-7ed0-49db-825d-657dfa3a6711.png)

**Reply by nyyssont** [reply]
In your first message, you express a concern that the white areas may not be missing data:

> The grains have some white areas in my data. I refer to the content in the " filling missing data", but it had no effect. I think my code is error, and the white area maybe not missing data.

So a logical first step here would be to verify if the white areas are missing data or not. Doing

´´´
plot(ebsd)
´´´

will accomplish this task. The plot will show if there are indexed pixels in the regions referred to here as white areas. These will then need to be set as "not indexed" to continue to filling them out with something else.

**Reply by Zhanlf0226** [reply]
thank you for your answer!

the result is like this
![image](https://user-images.githubusercontent.com/87469624/125776111-12f448d5-4177-42ec-90af-967e376401e4.png)
if I  reconstruct the grain , the phase plot is 
![image](https://user-images.githubusercontent.com/87469624/125777146-b2b186dd-4eee-4d82-b5e5-fb49feade0cf.png)
my code is 
![image](https://user-images.githubusercontent.com/87469624/125777634-f6ebd0c1-05db-479b-80e0-246745438c4b.png)
the result is satisfactory.

but in the orientation plot, i reconstruct the grain or denoise/ fill the missing area, it can't run the code. and it has a warning
![image](https://user-images.githubusercontent.com/87469624/125778194-4b83844a-c189-4871-ac6b-7168e78fcf85.png)
The plots and code are in above.

**Comment by zmichels** [comment, marked as answer]
Try
plot(ebsd_smoothed) 

I think the trouble is you are making a “smoothed” EBSD variable, but you don’t plot it at the end, you plot the original one again.

**Reply by Zhanlf0226** [reply]
thanks！
